_core/smart/neon-node.md

```diff
@@ -68,6 +68,78 @@ INFO [1588108047.585] [src/chainstate/stacks/index/marf.rs:732] First-ever block
 
+### Running a miner
+
+Once you've followed the above steps to run a node, it's only a few more steps to run a Proof-of-burn miner on the Neon testnet.
+
+First, we need to generate a keychain. With this keychain, we'll get some testnet BTC from a faucet, and then use that BTC to start mining.
+
+To get a keychain, the simplest way is to use the `blockstack-cli`. We'll use the `make_keychain` command, and pass `-t` to indicate that we want a testnet keychain.
+
+```bash
+npx blockstack-cli@1.1.0-beta.1 make_keychain -t
+```
+
+After this runs, you should see some output that looks like this:
+
+```json
+{
+  "mnemonic": "exhaust spin topic distance hole december impulse gate century absent breeze ostrich armed clerk oak peace want scrap auction sniff cradle siren blur blur",
+  "keyInfo": {
+    "privateKey": "2033269b55026ff2eddaf06d2e56938f7fd8e9d697af8fe0f857bb5962894d5801",
+    "address": "STTX57EGWW058FZ6WG3WS2YRBQ8HDFGBKEFBNXTF",
+    "btcAddress": "mkRYR7KkPB1wjxNjVz3HByqAvVz8c4B6ND",
+    "index": 0
+  }
+}
+```
+
+We need to get some testnet BTC to that address. Grab the `btcAddress` field, and head over to [the Stacks testnet website](https://testnet.blockstack.org/faucet). In the BTC faucet section, past in your `btcAddress`, and submit. You'll be sent 0.5 testnet BTC to that address. Don't lose this information - we'll need to use the `privateKey` field later on.
+
+Now, we need to configure out node to use this Bitcoin keychain. In the `stacks-blockchain` folder, create a new file called `testnet/stacks-node/conf/neon-miner-conf.toml`.
+
+Paste in the following configuration:
+
+```toml
+[node]
+rpc_bind = "0.0.0.0:20443"
+p2p_bind = "0.0.0.0:20444"
+bootstrap_node = "048dd4f26101715853533dee005f0915375854fd5be73405f679c1917a5d4d16aaaf3c4c0d7a9c132a36b8c5fe1287f07dad8c910174d789eb24bdfb5ae26f5f27@neon.blockstack.org:20444"
+# Enter your private key here!
+seed = "replace-with-your-private-key"
+miner = true
+
+[burnchain]
+chain = "bitcoin"
+mode = "neon"
+peer_host = "neon.blockstack.org"
+rpc_port = 18443
+peer_port = 18444
+
+[[mstx_balance]]
+address = "STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6"
+amount = 10000000000000000
+[[mstx_balance]]
+address = "ST11NJTTKGVT6D1HY4NJRVQWMQM7TVAR091EJ8P2Y"
+amount = 10000000000000000
+[[mstx_balance]]
+address = "ST1HB1T8WRNBYB0Y3T7WXZS38NKKPTBR3EG9EPJKR"
+amount = 10000000000000000
+[[mstx_balance]]
+address = "STRYYQQ9M8KAF4NS7WNZQYY59X93XEKR31JP64CP"
+amount = 10000000000000000
+```
+
+Now, grab your `privateKey` from earlier, when you ran the `make_keychain` command. Replace the `seed` field with your private key. Save and close this configuration file.
+
+To run your miner, run this in the command line:
+
+```bash
+stacks-node start config=./testnet/stacks-node/conf/neon-miner-conf.toml
```

Comment:
Should be `--config`. I'm getting:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: MissingOption(Keys(["--config", ""]))', testnet/stacks-node/src/main.rs:52:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I'm also getting a folder error because I went into the folder to edit the file there.. would urge us to add folder commands in here or say from which folder to run this

Reply:
In the correct folder, I'm now getting:

```
stacks-blockchain on  master [?] is 📦 v0.0.1 via 🦀 v1.42.0 
❯ stacks-node start --config=./testnet/stacks-node/conf/neon-miner-conf.toml

==> ./testnet/stacks-node/conf/neon-miner-conf.toml
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: Unexpected('`'), line: Some(26), col: 26, at: Some(861), message: "", key: [] } }', testnet/stacks-node/src/config.rs:37:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
had a typo at the end of the toml file. Now the node seems to run with only one output so far:
```
❯ stacks-node start --config=./testnet/stacks-node/conf/neon-miner-conf.toml

==> ./testnet/stacks-node/conf/neon-miner-conf.toml
INFO [1588972789.042] [src/chainstate/stacks/index/marf.rs:765] First-ever block 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206
```

A confirmation in the logs would be nice! Not seeing any difference to the node running logs